SPECS/cni/cni.spec

```diff
@@ -65,13 +66,13 @@ the container is deleted. Because of this focus, CNI has a wide
 range of support and the specification is simple to implement.
 
 %prep
-%setup -q
+%autosetup -N
 cp %{SOURCE2} build.sh
-
-%build
 # create vendor folder from the vendor tarball and set vendor mode
 tar -xf %{SOURCE3} --no-same-owner
+%patch 0 -p1 -d "vendor/golang.org/x/text"
```

Comment:
Instead of using `-d` I'd suggest modifying the paths in the patch. This way we can use `%autosetup` or `%autopatch` to make adding or removing patches easier.

Reply:
Checkout https://github.com/microsoft/azurelinux/blob/04f7ddc43f8949db24791413eba21172d11b925e/SPECS/coredns/coredns.spec#L46

Looks like the auto patcher has adopted the pattern we used elsewhere (its mostly gone from 3.0, probably because we updated all the packages, but if you look for "# Apply vendor before patching" in 2.0 you can see quite a few).

Reply:
It should be enough to just change the paths in the patch file, or you can do it "properly" and use `git format-patch`
 to update the patch file. For these vendor ones I just tend to manually edit the patch file.

Reply:
So, to clarify, should I do something along the lines of 
`%autosetup -a 3 -p1` and modify the patch file, or should I do something like `%autosetup -N` and keep the `tar` / replace `%patch` with `%autopatch`?

Reply:
It sounds like the former.